src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -670,6 +670,32 @@ private void FlushPendingOutput()
                         }
                     }
 
+                    //create the file in the main thread and flush the contents in the background thread
+                    // In case , there is an error in file creation, the logic would return from here and
+                    // not consume the exception
+                    try
+                    {
+                        string baseDirectory = Path.GetDirectoryName(transcript.Path);
+                        if(baseDirectory.Equals(transcript.Path.TrimEnd(System.IO.Path.DirectorySeparatorChar)))
+                        {
+                            Exception exception = PSTraceSource.NewArgumentException(transcript.Path,
+                            InternalHostUserInterfaceStrings.InvalidTranscriptFilePath, transcript.Path);
+                            throw exception;
+                        }
+                        if(!Directory.Exists(baseDirectory))
+                        {
+                            Directory.CreateDirectory(baseDirectory);
+                        }                        
+                        if(!File.Exists(transcript.Path))
+                        {
+                            using (File.Create(transcript.Path)){}
```

Comment:
The filestream object will be null only when File creation fails which is recorded as an exception. 
So, it seems safe to use dispose directly. Made this change.